llama_deploy/message_queues/solace/solace.py

```diff
@@ -0,0 +1,272 @@
+"""Solace Message Queue."""
+
+import asyncio
+import json
+import time
+import threading
+from string import Template
+from logging import getLogger
+from typing import Any, Dict, List, Literal, TYPE_CHECKING
+
+from pydantic import BaseModel, Field
+from pydantic_settings import BaseSettings, SettingsConfigDict
+
+from llama_deploy.message_queues.base import BaseMessageQueue
+from llama_deploy.messages.base import QueueMessage
+from llama_deploy.message_consumers.base import BaseMessageQueueConsumer, StartConsumingCallable
+
+# if TYPE_CHECKING:
+from solace.messaging.errors.pubsubplus_client_error import PubSubPlusClientError, IllegalStateError
+from solace.messaging.resources.topic_subscription import TopicSubscription
+from solace.messaging.resources.topic import Topic
+from solace.messaging.publisher.persistent_message_publisher import PersistentMessagePublisher, \
+    MessagePublishReceiptListener
+from solace.messaging.receiver.message_receiver import MessageHandler, InboundMessage
+from solace.messaging.receiver.persistent_message_receiver import PersistentMessageReceiver
+from solace.messaging.config.retry_strategy import RetryStrategy
+from solace.messaging.messaging_service import MessagingService
+from solace.messaging.resources.queue import Queue
+from solace.messaging.config.missing_resources_creation_configuration import MissingResourcesCreationStrategy
+
+from .boot import Boot
+
+# Constants
+MAX_SLEEP = 10
+QUEUE_TEMPLATE = Template('Q/$iteration')
+lock = threading.Lock()
+
+# Configure logger
+logger = logger = getLogger(__name__)
+
+class MessagePublishReceiptListenerImpl(MessagePublishReceiptListener):
+    """Message publish receipt listener for Solace message queue."""
+    def __init__(self):
+        self._publish_count = 0
+        pass
+
+    @property
+    def get_publish_count(self):
+        return self._publish_count
+
+    def on_publish_receipt(self, publish_receipt: 'PublishReceipt'):
+        with lock:
```

Comment:
This is a global level lock used, for a class variable.
Should we update the scope of the lock to be a class variable as well?